homeassistant/components/traccar_server/helpers.py

```diff
@@ -22,3 +22,15 @@ def get_first_geofence(
         (geofence for geofence in geofences if geofence["id"] in target),
         None,
     )
+
+def get_geofence_ids(
+    device: DeviceModel,
+    position: PositionModel,
+) -> list[int]:
+    # For Traccar <5.8
+    if "geofenceIds" in device:
+        return device["geofenceIds"]
+    # For Traccar >=5.8 https://github.com/traccar/traccar/commit/30bafaed42e74863c5ca68a33c87f39d1e2de93d
+    elif "geofenceIds" in position:
+        return position["geofenceIds"]
```

Comment:
Switch places of these, we assume most people are running current versions.
Also, there is no need for elif after return, just do another if.